### PR TITLE
Add metric support for grpc

### PR DESCRIPTION
**Pull request by jack-berg**
Was surprised to learn that gRPC metrics weren't already enabled. Any reason not to enable them?

### CODE REVIEW

File: testing-common/src/main/java/io/opentelemetry/instrumentation/testing/LibraryTestRunner.java [hunk 1]
```diff
@@ -44,7 +46,7 @@ public final class LibraryTestRunner extends InstrumentationTestRunner {
     GlobalOpenTelemetry.resetForTest();
 
     testSpanExporter = InMemorySpanExporter.create();
-    testMetricExporter = InMemoryMetricExporter.create();
+    testMetricExporter = InMemoryMetricExporter.create(AggregationTemporality.DELTA);
```
Comment (jack-berg) [inline testing-common/src/main/java/io/opentelemetry/instrumentation/testing/LibraryTestRunner.java:49]: Cumulative metrics are a bad fit for tests. We have a shard instance of the SDK for all tests. If temporality is cumulative, then metrics that initially received measurements for one test will continue to report for the remaining tests. The result is very cumbersome to assert against and debug.

File: instrumentation/grpc-1.6/testing/src/main/java/io/opentelemetry/instrumentation/grpc/v1_6/AbstractGrpcStreamingTest.java [hunk 2]
```diff
@@ -214,6 +216,56 @@ public void onCompleted() {
...
+                metrics.anySatisfy(
+                    metric ->
+                        MetricAssertions.assertThat(metric)
+                            .hasUnit("ms")
+                            .hasDoubleHistogram()
+                            .points()
+                            .anySatisfy(
+                                point ->
+                                    MetricAssertions.assertThat(point)
+                                        .hasAttributes(
```
[the remark this answers is not in the document]
Comment (jack-berg) [reply instrumentation/grpc-1.6/testing/src/main/java/io/opentelemetry/instrumentation/grpc/v1_6/AbstractGrpcStreamingTest.java:233]: No we don't have that. But this attributes assertion is an exact match.

[the remark this answers is not in the document]
Comment (trask) [reply instrumentation/grpc-1.6/testing/src/main/java/io/opentelemetry/instrumentation/grpc/v1_6/AbstractGrpcStreamingTest.java:233]: oh, good, and I found the `MetricsView` which filters down to the low cardinality attributes 👍

**Review by trask — APPROVED** [review]
📈

**Review by mateuszrzeszutek — APPROVED** [review]

File: testing/agent-exporter/src/main/java/io/opentelemetry/javaagent/testing/exporter/OtlpInMemoryMetricExporter.java [hunk 3]
```diff
@@ -34,7 +34,7 @@ void reset() {
 
   @Override
   public AggregationTemporality getAggregationTemporality(InstrumentType instrumentType) {
-    return AggregationTemporality.CUMULATIVE;
+    return AggregationTemporality.DELTA;
```
Comment (laurit) [inline testing/agent-exporter/src/main/java/io/opentelemetry/javaagent/testing/exporter/OtlpInMemoryMetricExporter.java:37]: @jack-berg this seems to make a micrometer test a bit flaky https://ge.opentelemetry.io/s/mwt7qlr5nwqig/tests/:instrumentation:micrometer:micrometer-1.5:javaagent:testBaseTimeUnit/io.opentelemetry.javaagent.instrumentation.micrometer.v1_5.TimerSecondsTest/testTimerWithBaseUnitSeconds()?expanded-stacktrace=WyIwIl0&top-execution=1
To reproduce insert `Thread.sleep(1_000);` between `timer.record()` calls. Any ideas where the issue is?

**Pull request merged**